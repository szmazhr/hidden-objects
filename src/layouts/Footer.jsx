import Favorite from '@mui/icons-material/Favorite';
import GitHub from '@mui/icons-material/GitHub';

const project = 'hidden-objects'; // change in every project

function Footer() {
  return (
    <footer className="main-footer">
      <FooterText />
      <FooterIcons />
    </footer>
  );
}

export default Footer;

/* ------------- Local Components -------------*/

const name = 'Shahzar Mazhar';
const username = 'szmazhr';
const profileLink = `https://github.com/${username}`;

function FooterText() {
  return (
    <div className="footer-text">
      <p>
        Made with
        <Favorite color="error" />
        by
        <a href={profileLink}>{name}</a>
      </p>
    </div>
  );
}

function FooterIcons() {
  return (
    <div className="footer-icons">
      <a href={`${profileLink}/${project}`}>
        <GitHub />
      </a>
    </div>
  );
}
