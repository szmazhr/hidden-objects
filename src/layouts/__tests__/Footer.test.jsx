import { render } from '@testing-library/react';
import Footer from '../Footer';

jest.mock('@mui/icons-material/Favorite', () => () => ' FavoriteIcon ');
jest.mock('@mui/icons-material/GitHub', () => () => 'GitHubIcon');

describe('Footer', () => {
  it('should render correctly', () => {
    const { asFragment } = render(<Footer />);
    expect(asFragment()).toMatchSnapshot();
  });
});
