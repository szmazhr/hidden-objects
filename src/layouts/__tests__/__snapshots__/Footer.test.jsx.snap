// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer should render correctly 1`] = `
<DocumentFragment>
  <footer
    class="main-footer"
  >
    <div
      class="footer-text"
    >
      <p>
        Made with FavoriteIcon by
        <a
          href="https://github.com/szmazhr"
        >
          Shahzar Mazhar
        </a>
      </p>
    </div>
    <div
      class="footer-icons"
    >
      <a
        href="https://github.com/szmazhr/hidden-objects"
      >
        GitHubIcon
      </a>
    </div>
  </footer>
</DocumentFragment>
`;
